ipv8/community.py

```diff
@@ -180,40 +139,23 @@ def on_deprecated_message(self, source_address, data):
                             self.deprecated_message_names[data[22]], *source_address)
 
     def bootstrap(self):
-        if time() - self.last_bootstrap < BOOTSTRAP_TIMEOUT:
-            return
-        self.logger.debug("Bootstrapping %s, current peers %d", self.__class__.__name__, len(self.get_peers()))
-        self.last_bootstrap = time()
-        for socket_address in _DEFAULT_ADDRESSES:
-            self._ensure_blacklisted(socket_address)
-            self.walk_to(socket_address)
-
-    def _ensure_blacklisted(self, address):
+        if not self._bootstrappers_initialized:
+            self.bootstrappers = [bootstrapper for bootstrapper in self.bootstrappers if bootstrapper.initialize(self)]
+            self._bootstrappers_initialized = True
+        for bootstrapper in self.bootstrappers:
+            task = self.register_anonymous_task(f"bootstrap {repr(bootstrapper)}", bootstrapper.get_addresses,
+                                                self, 60.0)
+            task.add_done_callback(self.received_bootstrapped_addresses)
+
+    def received_bootstrapped_addresses(self, addresses):
```

Comment:
I would be surprised if this is working. If I remember correctly, the done callback takes a `Future` as an argument. Casting a finished `Future` to a `list` always seems to yield an empty `list`.

Reply:
Woops, that should indeed be `addressses.result()`.